WordPress/src/main/java/org/wordpress/android/ui/JetpackConnectionDeeplinkActivity.java

```diff
@@ -91,48 +88,35 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     @Override
-    protected void onStop() {
-        mDispatcher.unregister(this);
-        super.onStop();
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (requestCode == RequestCodes.JETPACK_LOGIN && resultCode == RESULT_OK) {
+            trackResult();
+        }
+        finishAndGoBackToSource();
     }
 
     @Override
-    protected void onStart() {
-        super.onStart();
-        mDispatcher.register(this);
+    public void onBackPressed() {
+        super.onBackPressed();
+        finishAndGoBackToSource();
     }
 
-    @Override
-    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
-        super.onActivityResult(requestCode, resultCode, data);
-        if (requestCode == RequestCodes.JETPACK_LOGIN && resultCode == RESULT_OK && mSource == Source.STATS) {
-            showStats();
+    private void trackResult() {
+        if (!TextUtils.isEmpty(mReason)) {
```

Comment:
I think you mentioned that `mReason` could be non-empty but it might still be successful since it might be "already-connected". I don't think this code is currently handling that.

Reply:
This is just tracking and toasting the reason but still handling the callback back to the original activity.

Reply:
I am not sure what you meant in this comment, but the changes you made looks good to me.